### PR TITLE
Clarifies notation and units in Ensemble Kalman Inversion documentation

### LINKED ISSUE #46: Clarify notation and units in Ensemble Kalman Inversion documentation?
A few clarifications, definitions of symbols, and perhaps units in the [docs on Ensemble Kalman Inversion](https://clima.github.io/EnsembleKalmanProcesses.jl/dev/ensemble_kalman_inversion/#Ensemble-Kalman-Inversion) might improve the docs-reading experience. The docs now read

image

For example, the meaning of "curly N"

image

it's parameters, or the units of the covariance matrix are not specified. This leads to questions about whether the parameter update formula has the correct dimensions. The inner produce <., .> is also not defined.

### PR DESCRIPTION
**Pull request by navidcy**
Attempts to resolve #46.

### CODE REVIEW

**Comment by navidcy** [comment]
@glwagner and @adelinehillier, how does it read now? Feel free to edit at your liking.

**Comment by glwagner** [comment]
@navidcy do we have docs previews?

**Review by glwagner — APPROVED** [review]

**Comment by glwagner** [comment]
Thanks @navidcy, I appreciate the extra verbosity.

**Comment by navidcy** [comment]
> Is Gamma_y a scalar variance or a covariance matrix (eg curly N is the multivariate Gaussian)... ?

Hm, didn't think of that. 
Still Γ is positive definite and thus the inverse square root makes sense, right?

**Comment by navidcy** [comment]
Even if it's a matrix, it should be a positive definite one. The inverse square root is then, nothing else, from the eigen decomposition of Γ but with the inverse square root of the eigenvalues of Γ in the diagonal matrix. Right?

**Comment by navidcy** [comment]
But perhaps it's simpler if we go back to Γ^{-1}. :)

**Comment by glwagner** [comment]
> Even if it's a matrix, it should be a positive definite one. The inverse square root is then, nothing else, from the eigen decomposition of Γ but with the inverse square root of the eigenvalues of Γ in the diagonal matrix. Right?

Ah ok.

> But perhaps it's simpler if we go back to Γ^{-1}. :)

Your call.

**Comment by navidcy** [comment]
> > Even if it's a matrix, it should be a positive definite one. The inverse square root is then, nothing else, from the eigen decomposition of Γ but with the inverse square root of the eigenvalues of Γ in the diagonal matrix. Right?
> 
> Ah, I see.

Well, now that I'm thinking about it, same questions arise if Γ is a matrix and we write Γ^{-1}. Does the inverse exist and is it well-defined? So I vote for keeping the symmetric version and adding a clarification, I'll do that. :)

**Comment by glwagner** [comment]
> > > Even if it's a matrix, it should be a positive definite one. The inverse square root is then, nothing else, from the eigen decomposition of Γ but with the inverse square root of the eigenvalues of Γ in the diagonal matrix. Right?
> > 
> > 
> > Ah, I see.
> 
> Well, now that I'm thinking about it, same questions arise if Γ is a matrix and we write Γ^{-1}. Does the inverse exist and is it well-defined? So I vote for keeping the symmetric version and adding a clarification, I'll do that. :)

Okay. Also need to clarify that Gamma_y is a covariance matrix rather than variance and N is the multivariate Gaussian distribution. It could be nice to remark on the meaning of Gamma_y too (represents uncertainty, including correlated uncertainty, in the observations?)

**Comment by navidcy** [comment]
@glwagner have a look at 46b0969 and feel free to rephrase to make it even clearer! :)

File: docs/src/ensemble_kalman_inversion.md [hunk 1]
```diff
@@ -1,37 +1,56 @@
...
+One of the ensemble Kalman processes implemented in `EnsembleKalmanProcesses.jl` is the ensemble Kalman inversion ([Iglesias et al, 2013](http://dx.doi.org/10.1088/0266-5611/29/4/045001)). The ensemble Kalman inversion (EKI) is a derivative-free ensemble optimization method that seeks to find the optimal parameters ``\theta \in \mathbb{R}^p`` in the inverse problem
 
-\\[ y = \mathcal{G}(\theta) + \eta, \\]
+```math
+y = \mathcal{G}(\theta) + \eta,
+```
+
+where ``\mathcal{G}`` denotes the forward map, ``y \in \mathbb{R}^d`` is the vector of observations
+and ``\eta  \in \mathbb{R}^d`` is additive Gaussian observational noise. Here, we take 
+``\eta \sim \mathcal{N}(0, \Gamma_y)`` from a multivariate ``d``-dimensional normal distribution
```
Comment (odunbar) [inline docs/src/ensemble_kalman_inversion.md:11]: You say eta is Gaussian twice. Also it is more common to say d-dimensional multivariate normal rather than multivariate ``d``-dimensional normal.

File: docs/src/ensemble_kalman_inversion.md [hunk 2]
```diff
@@ -1,37 +1,56 @@
...
+with zero mean and covariance matrix ``\Gamma_y``.  This noise structure aims to represent the
+uncertainty, including correlated uncertainty, in the observations. Note that ``p`` is the
+size of the parameter vector ``\theta`` and ``d`` is taken to be the size of the observation
+vector ``y``.
```
Comment (odunbar) [inline docs/src/ensemble_kalman_inversion.md:15]: repetition of definition p and d? perhaps just write this out when y and theta are defined?

"uncertainty, including correlated uncertainty, in the observations" -> "correlations between observations"

File: docs/src/ensemble_kalman_inversion.md [hunk 3]
```diff
@@ -1,37 +1,56 @@
...
 
-where $\mathcal{G}$ denotes the forward map, $y \in \mathbb{R}^d$ is the vector of observations and $\eta \sim \mathcal{N}(0, \Gamma_y)$ is additive Gaussian observational noise. Note that $p$ is the size of the parameter vector $\theta$ and $d$ is taken to be the size of the observation vector $y$. The EKI update equation for parameter vector $\theta^{(j)}$ of ensemble member $j$ is
+The parameter vector of the ``j``-th ensemble member at the ``n``-th iteration is ``\theta^{(j)}_n``. The EKI update equation for parameter vector ``\theta^{(j)}`` is
 
 ```math
-\theta_{n+1}^{(j)} = \theta_{n}^{(j)} - \dfrac{\Delta t_n}{J}\sum_{k=1}^J\langle \mathcal{G}(\theta_n^{(k)}) - \bar{\mathcal{G}}_n, \Gamma_y^{-1}(\mathcal{G}(\theta_n^{(j)}) - y) \rangle \theta_{n}^{(k)},
+\theta_{n+1}^{(j)} = \theta_{n}^{(j)} - \dfrac{\Delta t_n}{J}\sum_{k=1}^J \left \langle \frac{\mathcal{G}(\theta_n^{(k)}) - \bar{\mathcal{G}}_n}{\sqrt{\Gamma_y}} \, , \, \frac{\mathcal{G}(\theta_n^{(j)}) - y}{\sqrt{\Gamma_y}} \right \rangle \theta_{n}^{(k)} ,
```
Comment (odunbar) [inline docs/src/ensemble_kalman_inversion.md:20]: We do not use the square root notation of the matrix in implementation (this implies one needs to factorize, compute the eigenvalues, and square root them, which we don't do), I would keep it as Gamma^{-1} on the second term - this is also in keeping with all statistical methods papers which are based on Gaussian methods

File: docs/src/ensemble_kalman_inversion.md [hunk 4]
```diff
@@ -1,37 +1,56 @@
...
 ```math
-\bar{\mathcal{G}}_n = \dfrac{1}{J}\sum_{k=1}^J\mathcal{G}(\theta_n^{(k)}).
+\bar{\mathcal{G}}_n = \dfrac{1}{J}\sum_{k=1}^J\mathcal{G}(\theta_n^{(k)}) ,
 ```
 
-The EKI algorithm is considered converged when the ensemble achieves sufficient consensus/collapse in parameter space. The final estimate $\bar{\theta}_{N_{it}}$ is taken to be the ensemble mean at the final iteration,
+and angle brackets denote the Euclidean inner product. The covariance matrix ``\Gamma_y`` is
+positive-definite and, therefore, raising it to the power ``-1/2`` is meaningful. By normalizing
+with ``\Gamma_y^{-1/2}`` we render the elements inside the inner product non-dimensional.
+
```
Comment (odunbar) [inline docs/src/ensemble_kalman_inversion.md:34]: Remove the positive definiteness and -1/2 stuff. We can keep the mention of nondimensionalization, I think this is relevant to user understanding.

File: docs/src/ensemble_kalman_inversion.md [hunk 5]
```diff
@@ -64,18 +83,19 @@ A call to the inversion algorithm can be performed with the `update_ensemble!` f
 
 A typical use of the `update_ensemble!` function given the ensemble Kalman inversion object `ekiobj`, the dynamical model `Ψ` and the observation map `H` is
 ```julia
-N_iter = 20 # Number of steps of the algorithm
+N_iterations = 20 # Number of steps of the algorithm
```
Comment (odunbar) [inline docs/src/ensemble_kalman_inversion.md:86]: We have a glossary of terms for consistency amongst documentation and code for the inverse problem examples.

https://clima.github.io/EnsembleKalmanProcesses.jl/dev/glossary/

Though it is fine to change `N_iter` to `N_iteration` if you believe it is needed, let's do it in a separate PR, as it needs to be changed everywhere.

**Review by odunbar — COMMENTED** [review]
Tahnks! A few small comments. 

RE notation - we have a[ glossary  ](https://clima.github.io/EnsembleKalmanProcesses.jl/dev/glossary/) which links notation in code and notes. We have tried to be consistent here, so changing these will require its own PR. as things will need to change everywhere.

RE the formulas, the notation we use matches papers that describe the algorithms as well as how they are implemented, so i think we should leave these.

**Comment by navidcy** [comment]
> RE notation - we have a[ glossary ](https://clima.github.io/EnsembleKalmanProcesses.jl/dev/glossary/) which links notation in code docs and the application examples. We have tried to be consistent here, so changing these will require its own PR. as things will need to change everywhere.

Perhaps raise this as an issue (if it's not already there)?

File: docs/src/ensemble_kalman_inversion.md (hunk 5 above)
Comment (navidcy) [reply docs/src/ensemble_kalman_inversion.md:86]: It's not *needed*, but what is really needed? We can get philosophical here. It's just much easier to read code if it's verbose and here 4-5 characters make the semi-incomprehensible `N_iter` to read almost English. :)

**Comment by navidcy** [comment]
preview @ https://clima.github.io/EnsembleKalmanProcesses.jl/previews/PR47/ensemble_kalman_inversion/

**Comment by navidcy** [comment]
@odunbar I did everything, do you want to have another look?

File: docs/src/ensemble_kalman_inversion.md [hunk 6]
```diff
@@ -1,37 +1,58 @@
...
+across ensemble members,
 
 ```math
-\bar{\mathcal{G}}_n = \dfrac{1}{J}\sum_{k=1}^J\mathcal{G}(\theta_n^{(k)}).
+\bar{\mathcal{G}}_n = \dfrac{1}{J}\sum_{k=1}^J\mathcal{G}(\theta_n^{(k)}) ,
 ```
 
-The EKI algorithm is considered converged when the ensemble achieves sufficient consensus/collapse in parameter space. The final estimate $\bar{\theta}_{N_{it}}$ is taken to be the ensemble mean at the final iteration,
+and angle brackets denote the Euclidean inner product. By multiplying with ``\Gamma_y^{-1}``
+(which has dimensions ``y^2``) we render the inner product non-dimensional.
```
Comment (odunbar) [inline docs/src/ensemble_kalman_inversion.md:35]: I think you mean ``d \times d`` rather than ``y^2``

**Review by odunbar — APPROVED** [review]
I'm happy except for the typo, LGTM, thanks Navid!

PS you can view the docs by going the the bottom of the thread, and  click `Details` in the `documenter/deploy — Documentation build succeeded ` section

File: docs/src/ensemble_kalman_inversion.md (hunk 6 above)
Comment (navidcy) [reply docs/src/ensemble_kalman_inversion.md:35]: I actually meant "units of y^2". Thanks.

Comment (navidcy) [reply docs/src/ensemble_kalman_inversion.md:35]: But I'll just remove the parenthetical remark.
